stubs/cffi/_cffi_backend.pyi

```diff
@@ -0,0 +1,290 @@
+import sys
+import types
+from _typeshed import Incomplete, ReadableBuffer, WriteableBuffer
+from collections.abc import Callable, Hashable
+from typing import Any, ClassVar, Protocol, TypeVar, overload
+from typing_extensions import Literal, TypeAlias, final
+
+_T = TypeVar("_T")
+
+class _Allocator(Protocol):
+    def __call__(self, cdecl: str | CType, init: Any = ...) -> _CDataBase: ...
+
+FFI_CDECL: int
+FFI_DEFAULT_ABI: int
+RTLD_DEEPBIND: int
+RTLD_GLOBAL: int
+RTLD_LAZY: int
+RTLD_LOCAL: int
+RTLD_NODELETE: int
+RTLD_NOLOAD: int
+RTLD_NOW: int
+
+@final
+class CField:
+    bitshift: Incomplete
+    bitsize: Incomplete
+    flags: Incomplete
+    offset: Incomplete
+    type: Incomplete
+
+@final
+class CLibrary:
+    def close_lib(self, *args, **kwargs): ...
+    def load_function(self, *args, **kwargs): ...
+    def read_variable(self, *args, **kwargs): ...
+    def write_variable(self, *args, **kwargs): ...
+
+@final
+class CType:
+    abi: Incomplete
+    args: Incomplete
+    cname: Incomplete
+    elements: Incomplete
+    ellipsis: Incomplete
+    fields: Incomplete
+    item: Incomplete
+    kind: Incomplete
+    length: Incomplete
+    relements: Incomplete
+    result: Incomplete
+    def __dir__(self): ...
+
+@final
+class Lib:
+    def __delattr__(self, name): ...
+    def __dir__(self): ...
+    def __setattr__(self, name, value): ...
+
+@final
+class _CDataBase:
+    __name__: ClassVar[str] = ...
```

Comment:
Style nit: the `= ...` is usually unnecessary in stub files, and we prefer to omit it where possible

```suggestion
    __name__: ClassVar[str]
```